### DISCUSSION TITLE
Any way to install a module before compiling firmware?

### DISCUSSION THREAD

**Discussion by bjpirt** [micropython/micropython#11133] [Libraries & Drivers]
The title says it all really - I'm building firmware with a bunch of my own custom modules frozen in at build time and I was wonderingg if there's a way of installing an external module into the code before compiling. I had a search around but couldn't find anything obvious. Am I missing something? Obviously I could roll my own but wondered if there was anything built-in.

**Comment by Josverl** [comment]
There are a few ways documented on the regular docs, https://docs.micropython.org/en/latest/develop/extendingmicropython.html?highlight=external%20c%20module

Or are you looking for something else?

**Comment by bjpirt** [comment]
That's not quite what I'm looking for - probably easiest to give a concrete example. I'm building a binary image to flash to my device with all of my Python code baked in. I want to use the Microdot library so I want to install this into the source tree before building. If I were doing this at runtime it would be straightforward and well documented but I couldn't find any onfo on doing this at build time.

**Comment by Josverl** [comment]
Keyword: 'Frozen modules'
that can be included in your DIY firmware using a custom [port/board/manifest.py](https://docs.micropython.org/en/latest/reference/manifest.html#summary)

**Comment by bjpirt** [comment]
Thanks @Josverl
